src/vuex/modules/keystore.js

```diff
@@ -35,27 +35,30 @@ export default () => {
       return getSeed()
     },
     async getAddressFromSeed(store, { seedPhrase, network }) {
-      const wallet = await getWallet(seedPhrase, network, store)
+      const networkObject = await this.getNetworkInfo(network, store)
+      const wallet = await getWallet(seedPhrase, networkObject)
       return wallet.cosmosAddress
     },
-    async createKey(
-      { dispatch, state, getters },
-      { seedPhrase, password, name, network }
-    ) {
+    async getNetworkInfo(networkId, store) {
+      return store.getters.networks.find(({ id }) => id === networkId)
+    },
+    async createKey(store, { seedPhrase, password, name, network }) {
       // TODO extract the key storage from the key creation
       const { storeWallet } = await import("@lunie/cosmos-keys")
 
+      // get current network
+      const networkObject = await this.getNetworkInfo(network, store)
       // create a new key pair
-      const wallet = await getWallet(seedPhrase, network, { getters })
+      const wallet = await getWallet(seedPhrase, networkObject)
 
       storeWallet(wallet, name, password, network)
 
-      state.externals.track(`event`, `session`, `create-keypair`)
+      store.state.externals.track(`event`, `session`, `create-keypair`)
 
       // reload accounts as we just added a new one
-      dispatch("loadAccounts")
+      store.dispatch("loadAccounts")
```

Comment:
why the switch to `store`?

Reply:
well, I thought the store-free file should be just wallet.js.. But yeah, Polkadot account creation is still not working there. Maybe because of this?

Reply:
so I should remove store from the whole account creation process then

Reply:
all right. it's just weird as we in all the other parts destructure store to `{ commit, dispatch, getters }`